lib/Mojolicious/Controller.pm

```diff
@@ -242,12 +246,24 @@ sub session {
   my $stash = $self->stash;
   $self->app->sessions->load($self) unless exists $stash->{'mojo.active_session'};
 
-  # Hash
   my $session = $stash->{'mojo.session'} //= {};
+
+  # Require changed secret if session data will be set
+  my $set_operation = !!(@_ > 1 || ref $_[0]);
+  if (!$ENV{MOJO_ALLOW_INSECURE_SECRET} and (keys %$session or $set_operation)) {
+    my $app     = $self->app;
+    my $secret  = $app->secrets->[0];
+    my $moniker = $app->moniker;
+
+    Carp::croak 'Your secret passphrase must be changed to set session data (see FAQ for more)'
```

Comment:
I don't think these secrets are "passphrase"s (and the FAQ doesn't call them that either)